portability-api/src/main/java/org/datatransferproject/api/action/transfer/ReserveWorkerAction.java

```diff
@@ -0,0 +1,89 @@
+package org.datatransferproject.api.action.transfer;
+
+import com.google.api.client.util.Sleeper;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import org.datatransferproject.api.action.Action;
+import org.datatransferproject.api.action.ActionUtils;
+import org.datatransferproject.spi.cloud.storage.JobStore;
+import org.datatransferproject.spi.cloud.types.JobAuthorization;
+import org.datatransferproject.spi.cloud.types.PortabilityJob;
+import org.datatransferproject.types.client.transfer.ReserveWorker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import static org.datatransferproject.api.action.ActionUtils.decodeJobId;
+import static org.datatransferproject.spi.cloud.types.JobAuthorization.State.CREDS_AVAILABLE;
+import static org.datatransferproject.spi.cloud.types.JobAuthorization.State.CREDS_ENCRYPTION_KEY_GENERATED;
+
+/** Reserves a worker to process a transfer job. */
+public class ReserveWorkerAction implements Action<ReserveWorker, String> {
+    private static final Logger logger = LoggerFactory.getLogger(ReserveWorkerAction.class);
+
+    private final JobStore jobStore;
+
+    @Inject
+    ReserveWorkerAction(JobStore jobStore) {
+        this.jobStore = jobStore;
+    }
+
+    @Override
+    public Class<ReserveWorker> getRequestType() {
+        return ReserveWorker.class;
+    }
+
+    @Override
+    public String handle(ReserveWorker reserveWorker) {
+        String id = reserveWorker.getId();
```

Comment:
Could this be null? Should we validate?

Reply:
added validation (apparently Jackson does not honor the annotations we put in that say the field is required)